### ISSUE TITLE
[Bug] app (2.16alpha88) crashes on review change

### ISSUE THREAD

**Issue by criticalAY** [ankidroid/Anki-Android#12739]
###### Reproduction Steps

1. settings
2. reviewing
3. new card after review options to be changed

###### Expected Result
option set as marked/selected

###### Actual Result
app crashes

###### Debug info
AnkiDroid Version = 2.16alpha88

Android Version = 12

Manufacturer = Google

Model = sdk_gphone64_x86_64

Hardware = ranchu

Webview User Agent = Mozilla/5.0 (Linux; Android 12; sdk_gphone64_x86_64 Build/SE1A.220630.001; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/91.0.4472.114 Mobile Safari/537.36

ACRA UUID = 246da304-f39d-4cd2-9fc7-d1bc0bd83ad0

New schema = false

Scheduler = std2

Crash Reports Enabled = false

DatabaseV2 Enabled = true

- [ ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [ ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [ ] I have searched for similar existing issues here and on the user forum
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by criticalAY** [comment]
```
FATAL EXCEPTION: main
                                                                                                    Process: com.ichi2.anki, PID: 2087
                                                                                                    java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Integer
                                                                                                    	at com.ichi2.anki.preferences.ReviewingSettingsFragment$initSubscreen$1$1.invoke(ReviewingSettingsFragment.kt:41)
                                                                                                    	at com.ichi2.anki.preferences.ReviewingSettingsFragment$initSubscreen$1$1.invoke(ReviewingSettingsFragment.kt:40)
                                                                                                    	at com.ichi2.anki.preferences.PreferenceUtilsKt.setOnPreferenceChangeListener$lambda$0(PreferenceUtils.kt:30)
                                                                                                    	at com.ichi2.anki.preferences.PreferenceUtilsKt.$r8$lambda$v3IcnU4ch5B8dJg8Pj9zOe393os(Unknown Source:0)
                                                                                                    	at com.ichi2.anki.preferences.PreferenceUtilsKt$$ExternalSyntheticLambda0.onPreferenceChange(Unknown Source:2)
                                                                                                    	at androidx.preference.Preference.callChangeListener(Preference.java:1131)
                                                                                                    	at androidx.preference.ListPreferenceDialogFragmentCompat.onDialogClosed(ListPreferenceDialogFragmentCompat.java:109)
                                                                                                    	at androidx.preference.PreferenceDialogFragmentCompat.onDismiss(PreferenceDialogFragmentCompat.java:291)
                                                                                                    	at androidx.fragment.app.DialogFragment$3.onDismiss(DialogFragment.java:133)
                                                                                                    	at android.app.Dialog$ListenersHandler.handleMessage(Dialog.java:1408)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:201)
                                                                                                    	at android.os.Looper.loop(Looper.java:288)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7839)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
2022-10-28 23:39:06.346  2087-2087  UsageAnalytics          com.ichi2.anki                       D  sendAnalyticsException() description/fatal: java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Integer/true
2022-10-28 23:39:06.349  2087-2087  ACRA                    com.ichi2.anki                       E  ACRA caught a ClassCastException for com.ichi2.anki
                                                                                                    java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Integer
                                                                                                    	at com.ichi2.anki.preferences.ReviewingSettingsFragment$initSubscreen$1$1.invoke(ReviewingSettingsFragment.kt:41)
                                                                                                    	at com.ichi2.anki.preferences.ReviewingSettingsFragment$initSubscreen$1$1.invoke(ReviewingSettingsFragment.kt:40)
                                                                                                    	at com.ichi2.anki.preferences.PreferenceUtilsKt.setOnPreferenceChangeListener$lambda$0(PreferenceUtils.kt:30)
                                                                                                    	at com.ichi2.anki.preferences.PreferenceUtilsKt.$r8$lambda$v3IcnU4ch5B8dJg8Pj9zOe393os(Unknown Source:0)
                                                                                                    	at com.ichi2.anki.preferences.PreferenceUtilsKt$$ExternalSyntheticLambda0.onPreferenceChange(Unknown Source:2)
                                                                                                    	at androidx.preference.Preference.callChangeListener(Preference.java:1131)
                                                                                                    	at androidx.preference.ListPreferenceDialogFragmentCompat.onDialogClosed(ListPreferenceDialogFragmentCompat.java:109)
                                                                                                    	at androidx.preference.PreferenceDialogFragmentCompat.onDismiss(PreferenceDialogFragmentCompat.java:291)
                                                                                                    	at androidx.fragment.app.DialogFragment$3.onDismiss(DialogFragment.java:133)
                                                                                                    	at android.app.Dialog$ListenersHandler.handleMessage(Dialog.java:1408)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:201)
                                                                                                    	at android.os.Looper.loop(Looper.java:288)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7839)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
```

**Comment by criticalAY** [comment]
this was reported by some users in SDK31

**Comment by lukstbit** [comment]
Closed by #12740
@criticalAY Github tip: you can use [certain keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) in your PRs if you want to automatically close the issue that you fix.
